validator/project_validator.go

```diff
@@ -38,6 +38,7 @@ const (
 	DockerHostCreateTotalLimit              = 200
 	HostCreateLimitPerTask                  = 3
 	maxTaskSyncCommandsForDependenciesCheck = 300 // this should take about one second
+	GenerateSyntaxError                     = "project contains syntax error"
```

Comment:
I don't think this is referenced anywhere.

Reply:
Good catch -- originally the approach I was going to take was wrapping the errors we _didn't_ want to retry on, but I think there are actually more of those. If you think the existing implementation isn't strict enough (i.e. only retry on Save errors) we could switch back to this.